src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -485,6 +485,12 @@ public static MsBuildToolset GetMsBuildToolset(string userVersion, IConsole cons
 
             try
             {
+                //If "latest", take the default(highest) path
+                if (string.Equals(userVersion, "latest", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't think this does it.

If you look at line 502, this will still go into the path env var: https://github.com/NuGet/NuGet.Client/blob/0735174f6a7d01445d63a11df7cbc13d07e16353/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs#L495-L506